src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -48,10 +49,20 @@
     <DebugEngines>{2E36F1D4-B23C-435D-AB41-18E608940038}</DebugEngines>
   </PropertyGroup>
 
+  <ItemGroup>
+    <TestTargetFrameworkPath Include="$(TestPath)%(TestTargetFramework.Folder)"/>
+    <TestsSuccessfulSemaphore Include="@(TestTargetFrameworkPath->'%(FullPath)\$(MSBuildProjectFile).tests.lastsucceeded')"/>
+    <XunitResultsFile Include="@(TestTargetFrameworkPath->'%(FullPath)\$(XunitResultsFileName)')"/>
+  </ItemGroup>
+
   <!-- On command line run unit tests on CoreCLR after the Test target -->
   <Target Name="RunTestsForProject"
           AfterTargets="CheckTestCategories"
+          Inputs="@(PackagesConfig);$(TestRuntimePackageConfig);@(ReferenceCopyLocalPaths);@(Content);@(IntermediateAssembly);@(_DebugSymbolsIntermediatePath)"
+          Outputs="@(TestsSuccessfulSemaphore);@(XunitResultsFile)"
           Condition="'$(TestDisabled)'!='true'">
+		  
+    <CallTarget Targets="CopyTestToTestDirectory"/>
```

Comment:
Correct, dependencies are executed prior to evaluating the Inputs/Outputs of a target. This is true even if the dependency is expressed using Before/AfterTargets instead of DependsOnTargets. (At first this was annoying, but it makes sense: what if you wanted to intentionally affect the inputs of a task using a dependency?)

I am not generally a fan of using CallTarget, but it was the only reliable way I could find to achieve all of the objectives of #60. Another option would be to inline CopyTestToTestDirectory into RunTestsForProject, which is not crazy, but it would make the RunTestsForProject task quite large.